src/modules/identity/RadixRemoteIdentity.ts

```diff
@@ -0,0 +1,226 @@
+     * @returns A promise with a valid token to interact with the wallet
+     */
+    public static register(name: string, description: string, host = 'localhost', port = '54345'): Promise<string> {
+        return new Promise<string>((resolve, reject) => {
+            // This is an independant websocket because 'register' is a static method
+            const socket = new WebSocket(`ws:${host}:${port}`)
+            
+            socket.onopen = () => {
+                socket.send(JSON.stringify({
+                    jsonrpc: '2.0',
+                    method: 'register',
+                    params: {
+                        name,
+                        description,
+                        permissions: ['sign_atom', 'decrypt_ecies_payload', 'get_public_key'],
+                    },
+                    id: 0,
+                }))
+                socket.onmessage = (evt) => {
+                    const response = JSON.parse(evt.data)
+                    if (response.result) {
+                        resolve(response.result.token)
+                    } else {
+                        reject(response.error)
+                    }
+                }
+                socket.onerror = (error) => reject(new Error(error))
+            }
+        })
+    }
+
+    /**
+     * Returns the public key of this identity asynchronously
+     * 
+     * @param [host] - The host of the wallet
+     * @param [port] - The port in which the wallet server is being exposed
+     * @returns A promise with the public key of the identity
+     */
+    public static getRemotePublicKey(token, host = 'localhost', port = '54345'): Promise<Buffer> {
+        return new Promise<Buffer>((resolve, reject) => {
+            // This is an independant websocket because 'getRemotePublicKey' is a static method
+            const socket = new WebSocket(`ws:${host}:${port}`)
+            
+            socket.onopen = () => {
+                socket.send(JSON.stringify({
+                    jsonrpc: '2.0',
+                    method: 'get_public_key',
+                    params: { token },
+                    id: 0,
+                }))
+                socket.onmessage = (evt) => {
+                    const response = JSON.parse(evt.data)
+                    if (response.result) {
+                        resolve(response.result.data)
+                    } else {
+                        reject(response.error)
+                    }
+                }
+                socket.onerror = (error) => reject(new Error(error))
+            }
+        })
+    }
+
+    /**
+     * Determines whether the server is up or down
+     * 
+     * @param [host] - The host of the wallet
+     * @param [port] - The port in which the wallet server is being exposed
+     * @returns A promise with true or false whether the server is up or down
+     */
+    public static isServerUp(host = 'localhost', port = '54345'): Promise<Boolean> {
```

Comment:
I think you could resolve this as soon as the `onopen` function is triggered, it would be a lot faster than waiting for the timeout in the successful case.